src/gosling-track/gosling-track.ts

```diff
@@ -96,20 +111,6 @@ function GoslingTrack(HGC: import('@higlass/types').HGC, ...args: any[]): any {
 
         constructor(params: any[]) {
             const [context, options] = params;
-
-            // Check whether to load tabualr data-fetcher
-            if (usePrereleaseRendering(options.spec)) {
-                try {
-                    if (options.spec.data?.type === 'bam') {
-                        context.dataFetcher = new BamDataFetcher(HGC, context.dataConfig);
-                    } else if (options.spec.data?.type === 'vcf') {
-                        context.dataFetcher = new VcfDataFetcher(HGC, context.dataConfig);
-                    }
-                } catch (e) {
-                    console.warn('Error loading tabular data-fetcher', e);
-                }
-            }
```

Comment:
I wonder I why did not register these using `higlassRegister` before...🤷‍♂️

Reply:
I think it was because you needed to instantiate the `Worker` for each data loader. This was done in `gosling-track` constructor previously and the worker was stored on the track and accessed directly. Now the worker is hidden (within the data-fetcher) and we added an extra `getTabularData` fn which we inspect.